controllers/controller/devworkspacerouting/solvers/common_test.go

```diff
@@ -0,0 +1,125 @@
+package solvers
+
+import (
+	"testing"
+
+	"github.com/devfile/devworkspace-operator/apis/controller/v1alpha1"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetRouteForEndpointAnnotations(t *testing.T) {
+	tests := []struct {
+		name string
+
+		routingSuffix string
+		endpoint      v1alpha1.Endpoint
+		meta          DevWorkspaceMetadata
+		annotations   map[string]string
+
+		expectedAnnotationsKeys []string
+	}{
+		{
+			name: "nil",
+
+			annotations: nil,
+
+			expectedAnnotationsKeys: []string{
+				"controller.devfile.io/endpoint_name",
+				"haproxy.router.openshift.io/rewrite-target",
+			},
+		},
+		{
+			name: "empty",
+
+			annotations: map[string]string{},
+
+			expectedAnnotationsKeys: []string{
+				"controller.devfile.io/endpoint_name",
+				"haproxy.router.openshift.io/rewrite-target",
+			},
+		},
+		{
+			name: "defined",
```

Comment:
This test could be called: "Gets default OpenShift route annotation when annotations are defined".